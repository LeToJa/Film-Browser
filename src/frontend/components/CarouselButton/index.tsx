import { CarouselButtonProps } from './index.types'

export default function CarouselButton({
  onClick,
  disabled = false,
  children,
}: CarouselButtonProps) {
  return (
    <button onClick={onClick} disabled={disabled} className="carousel-button">
      {children}
    </button>
  )
}
