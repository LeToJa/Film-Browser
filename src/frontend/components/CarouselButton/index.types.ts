export type CarouselButtonProps = {
  onClick: () => void
  disabled?: boolean
  children: React.ReactNode
}
