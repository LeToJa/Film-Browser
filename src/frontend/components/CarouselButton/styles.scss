.carousel-button {
  padding: 0.5rem;
  border: none;
  background: transparent;
  color: black;
  font-weight: 900;
  cursor: pointer;
  border: 2px solid;
  border-radius: 3px;

  &:disabled {
    opacity: 0.5;
    cursor: not-allowed;
  }
}
