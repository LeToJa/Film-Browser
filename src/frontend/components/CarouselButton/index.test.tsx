import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { describe, it, expect, vi } from 'vitest'
import CarouselButton from '.'

describe('CarouselButton', () => {
  it('renders correctly with children', () => {
    const { asFragment } = render(
      <CarouselButton onClick={() => {}}>{'Next >'}</CarouselButton>
    )
    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByRole('button')).toHaveTextContent('Next >')
  })

  it('calls onClick when clicked', async () => {
    const user = userEvent.setup()
    const handleClick = vi.fn()
    render(<CarouselButton onClick={handleClick}>{'>'}</CarouselButton>)
    await user.click(screen.getByRole('button'))
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('is disabled when disabled prop is true', () => {
    const { asFragment } = render(
      <CarouselButton onClick={() => {}} disabled>
        {'>'}
      </CarouselButton>
    )
    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByRole('button')).toBeDisabled()
  })
})
