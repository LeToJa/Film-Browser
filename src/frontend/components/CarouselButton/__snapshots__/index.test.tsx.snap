// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CarouselButton > is disabled when disabled prop is true 1`] = `
<DocumentFragment>
  <button
    class="carousel-button"
    disabled=""
  >
    &gt;
  </button>
</DocumentFragment>
`;

exports[`CarouselButton > renders correctly with children 1`] = `
<DocumentFragment>
  <button
    class="carousel-button"
  >
    Next &gt;
  </button>
</DocumentFragment>
`;
